src/languages/nix.js

```diff
@@ -9,85 +9,291 @@ Category: system
 export default function(hljs) {
   const KEYWORDS = {
     keyword: [
-      "rec",
-      "with",
-      "let",
-      "in",
-      "inherit",
       "assert",
-      "if",
       "else",
-      "then"
+      "if",
+      "in",
+      "inherit",
+      "let",
+      "or",
+      "rec",
+      "then",
+      "with",
     ],
     literal: [
       "true",
       "false",
-      "or",
-      "and",
-      "null"
+      "null",
     ],
     built_in: [
-      "import",
+      // toplevel builtins
       "abort",
       "baseNameOf",
+      "builtins",
+      "derivation",
+      "derivationStrict",
       "dirOf",
+      "fetchGit",
+      "fetchMercurial",
+      "fetchTarball",
+      "fetchTree",
+      "fromTOML",
+      "import",
       "isNull",
+      "map",
+      "placeholder",
+      "removeAttrs",
+      "scopedImport",
+      "throw",
+      "toString",
+    ],
+  };
+
+  const BUILTINS = {
+    className: 'built_in',
+    match: new RegExp('(' + [
```

Comment:
Please use our regex built-in tools such as `concat`... so here you'd just wrap your map in `concat`.  And rememver it takes arguments, not an array, so we explode the array with `...`.

```
// something like
hljs.regex.concat(...LIST.map( /* add namespaces*/ ))
```